src/StreamDeckLib/ConnectionManager.cs

```diff
@@ -1,300 +1,360 @@
+		/// the <paramref name="token">cancellation token</paramref>.
+		/// </summary>
+		/// <param name="token">A cancellation token which can be used to break out of the wait loop.</param>
+		/// <returns>A value indicating whether a debugger was attached (<c>true</c>) or not (<c>false</c>).</returns>
+		private async Task<bool> AttachDebugger(CancellationToken token)
+		{
+			if (!_WaitForDebugger)
+			{
+				return true;
+			}
+
+			Console.Write("Attach a debugger to resume execution...");
+			
+			Debugger.Launch();
+
+			// Wait for a debugger to attach, but if the token gets a cancelation request, we need to exit.
+			while (!Debugger.IsAttached | !token.IsCancellationRequested)
+			{
+				// Polling every 1/2 second should be enough, rather than every 1/10th of a second.
+				// We could make this a parameter at a later time, but some validation would be needed
+				// to cap the minimum and maximum time allowed.
+				await Task.Delay(500, token); 
+			}
+
+			// If we exited because the token cancellation has been requested, return before doing any socket-related operations
+			return !token.IsCancellationRequested;
+		}
+		
+		
+        #region StreamDeck Methods
+
+        public async Task SetTitleAsync(string context, string newTitle)
+        {
+            var args = new SetTitleArgs()
+            {
+                context = context,
+                payload = new SetTitleArgs.Payload
+                {
+                    title = newTitle,
+                    TargetType = SetTitleArgs.TargetType.HardwareAndSoftware
+                }
+            };
+
+			await _Proxy.SendStreamDeckEvent(args);
+		}
+
+		public async Task SetImageAsync(string context, string imageLocation) {
+
+			var imgString = Convert.ToBase64String(File.ReadAllBytes(imageLocation), Base64FormattingOptions.None);
+
+			var args = new SetImageArgs
+			{
+				context = context,
+				payload = new SetImageArgs.Payload
+				{
+					TargetType = SetTitleArgs.TargetType.HardwareAndSoftware,
+					image = $"data:image/{new FileInfo(imageLocation).Extension.ToLowerInvariant().Substring(1)};base64, {imgString}"
+				}
+			};
+
+			await _Proxy.SendStreamDeckEvent(args);
+
+		}
+
+		public async Task ShowAlertAsync(string context)
+		{
+			var args = new ShowAlertArgs()
+			{
+				context = context
+			};
+			await _Proxy.SendStreamDeckEvent(args);
+		}
+
+		public async Task ShowOkAsync(string context)
+		{
+			var args = new ShowOkArgs()
+			{
+				context = context
+			};
+			await _Proxy.SendStreamDeckEvent(args);
+        }
+
+        public async Task SetSettingsAsync(string context, dynamic value)
+        {
+            var args = new SetSettingsArgs()
+            {
+                context = context,
+                payload = value
+            };
+			await _Proxy.SendStreamDeckEvent(args);
+        }
+
+		public async Task SetStateAsync(string context, int state)
+        {
+
+			var args = new SetStateArgs
+            {

```

Comment:
nit - these curlies are a little malformatted. just a nit, but if they could be more nicely/consistently formatted, more yay.

Reply:
No worries, I get it! I'm usually quite picky with indentation, brace placement, etc. I'd just merged my changes back into the changes made in the `dev` branch using Beyond Compare, assuming indentation/formatting wouldn't have changed. And now I just had a look at it in VS.... Queue Rick Flair audio of "Whooooo", but throw a "doggy" on the end of it! But kept my indentation changes to this block of code and 3 instances of curly braces on the wrong line.